src/parser/SpatialQuery.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024, University of Freiburg
+// Chair of Algorithms and Data Structures
+// Authors: Christoph Ullinger <ullingec@informatik.uni-freiburg.de>
+
+#include "parser/SpatialQuery.h"
+
+namespace parsedQuery {
+
+// ____________________________________________________________________________
+void SpatialQuery::addParameter(const SparqlTriple& triple) {
+  auto simpleTriple = triple.getSimple();
+  TripleComponent predicate = simpleTriple.p_;
+  TripleComponent object = simpleTriple.o_;
+
+  if (!predicate.isIri()) {
+    throw SpatialSearchException("Predicates must be IRIs");
+  }
+
+  std::string predString = predicate.getIri().toStringRepresentation();
+  if (predString.ends_with("left>")) {
+    setVariable("left", object, left_);
+  } else if (predString.ends_with("right>")) {
+    setVariable("right", object, right_);
+  } else if (predString.ends_with("nearestNeighbors>")) {
+    if (!object.isInt()) {
+      throw SpatialSearchException(
+          "The parameter 'nearestNeighbors' expects an integer (the maximum "
+          "number of nearest neighbors)");
+    }
+    maxResults_ = object.getInt();
+  } else if (predString.ends_with("maxDistance>")) {
+    if (!object.isInt()) {
+      throw SpatialSearchException(
+          "The parameter 'maxDistance' expects an integer (the maximum "
+          "distance in meters)");
+    }
+    maxDist_ = object.getInt();
+  } else if (predString.ends_with("bindDistance>")) {
+    setVariable("bindDistance", object, bindDist_);
+  } else if (predString.ends_with("algorithm>")) {
+    if (!object.isIri()) {
+      throw SpatialSearchException(
+          "The parameter 'algorithm' needs an IRI that selects the algorithm "
+          "to employ. Currently supported are 'spatialSearch:baseline' and "
+          "'spatialSearch:s2'.");
+    }
+    auto algoIri = object.getIri().toStringRepresentation();
+    if (algoIri.ends_with("baseline>")) {
+      algo_ = SpatialJoinAlgorithm::BASELINE;
+    } else if (algoIri.ends_with("s2>")) {
+      algo_ = SpatialJoinAlgorithm::S2_GEOMETRY;
```

Comment:
Before this PR we will probably merge the other PR by @Jonathan24680 , which introduces a third algorithm (based on boost::geometry).

Reply:
Adding another algorithm is not a problem even if we would merge this first. I changed all the "use baseline algorithm" booleans to this new enum which specifies the algorithm.